src/linear_algebra/bilinear_form.lean

```diff
@@ -174,6 +174,28 @@ end
 
 end bilin_form
 
+section left_right
+
+/-- The linear map obtained from a `bilin_form` by fixing the left co-ordinate and evaluating in
+the right. -/
+def bilin_form.right (A : bilin_form R M) (x : M) : M →ₗ[R] R :=
+{ to_fun := λ y, A x y,
+  map_add' := A.bilin_add_right x,
+  map_smul' := λ c, A.bilin_smul_right c x }
```

Comment:
I assume the motivation here is that unlike `to_lin`, this works for non-commutative `R`?

Reply:
Yes, I wonder if your proposed solution on #6585 would also work here to combine them?

Reply:
Yes, it definitely would - and I recently changed `linear_map.flip` to allow different rings, to allow swapping between the left and right version.

Reply:
OK, all the more reason for me to learn this simultaneous-`ℤ`-and-`R` trick!  I've fixed all the errors in #6585, can you perhaps demonstrate it there?

Reply:
@eric-wieser So far I addressed only this comment.  I've implemented `left` (now renamed `to_lin'`) as a weaker version of `to_lin`, using the same abbrevation-for-joint-generalization-trick as in #6585.  Then I've written a bilinear-map-specific `flip` and used `flip` + `to_lin` to replace the old `right`.  I believe `linear_map.flip` is not strong enough to use directly here.

This material will need to be spun off on its own PR, if we keep it, but before I do this please let me know if this approach works for what you had in mind.

Note that PR is semi-blocking #6427, which is itself a dependency of several other PRs, so let's try not to go too far down the rabbit hole.

Reply:
I'm in no rush, so long as the PRs don't totally rot.